app/models/assistantship.rb

```diff
@@ -0,0 +1,9 @@
+class Assistantship < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :course
+
+  validates :user,
+            presence: { message: 'does not exist' },
+            uniqueness: { scope: :course, message: 'is already an assistant for this course' }
+  validates :course, presence: true
+end
```

Comment:
This is fine for now but a more generic solution would be to have a table of `(user_id, course_id, role or permission)`. Now `Assistantship` and `Tearchership` do pretty much the same thing.